include/execution.hpp

```diff
@@ -599,23 +626,49 @@ namespace std::execution {
 
   /////////////////////////////////////////////////////////////////////////////
   // [execution.senders.queries], sender queries
-  template <__one_of<set_value_t, set_error_t, set_done_t> _CPO>
-    struct get_completion_scheduler_t {
-      template <sender _Sender>
-        requires tag_invocable<get_completion_scheduler_t, const _Sender&> &&
-          scheduler<tag_invoke_result_t<get_completion_scheduler_t, const _Sender&>>
-      auto operator()(const _Sender& __sndr) const noexcept
-          -> tag_invoke_result_t<get_completion_scheduler_t, const _Sender&> {
-        // NOT TO SPEC:
-        static_assert(
-          nothrow_tag_invocable<get_completion_scheduler_t, const _Sender&>,
-          "get_completion_scheduler<_CPO> should be noexcept");
-        return tag_invoke(*this, __sndr);
-      }
-    };
+  inline namespace __sender_queries {
+    namespace __impl {
+      template <__one_of<set_value_t, set_error_t, set_done_t> _CPO>
+        struct get_completion_scheduler_t {
+          template <sender _Sender>
+            requires tag_invocable<get_completion_scheduler_t, const _Sender&> &&
+              scheduler<tag_invoke_result_t<get_completion_scheduler_t, const _Sender&>>
+          auto operator()(const _Sender& __sndr) const noexcept
+              -> tag_invoke_result_t<get_completion_scheduler_t, const _Sender&> {
+            // NOT TO SPEC:
+            static_assert(
+              nothrow_tag_invocable<get_completion_scheduler_t, const _Sender&>,
+              "get_completion_scheduler<_CPO> should be noexcept");
+            return tag_invoke(*this, __sndr);
+          }
+        };
 
-  template <__one_of<set_value_t, set_error_t, set_done_t> _CPO>
-    inline constexpr get_completion_scheduler_t<_CPO> get_completion_scheduler{};
+      struct forwarding_sender_query_t {
+        template <class _Tag>
+        constexpr bool operator()(_Tag __tag) const noexcept {
+          if constexpr (tag_invocable<forwarding_sender_query_t, _Tag>) {
+            return tag_invoke(*this, (_Tag&&) __tag);
+          } else {
+            return false;
```

Comment:
Hmm - here we use a default of false which makes these customizations an includelist. That is inconsistent with the receiver queries cpo.

Reply:
Yes, I think false is the right default for senders, and true is the right default for receivers.